### PR TITLE
handleCardSetup

### PR DESCRIPTION
**Pull request by dweedon-stripe**
### Summary & motivation
* Adds support for `stripe.handleCardSetup`
* Rename PaymentIntents demo to intents
* Add demo for `handleCardSetup`
* Allow users to set a `STRIPE_PUBLISHABLE_KEY` environment variable to correspond with they `STRIPE_SECRET_KEY` they need to set for the `handleCardPayment` and `handleCardSetup` demos to work.

This is a slightly breaking change so it will be rolled into the 4.0.0 release.

### Testing & documentation
I wrote unit tests, updated the readme, and added a demo.

### CODE REVIEW

File: src/components/inject.js [hunk 1]
```diff
@@ -294,31 +294,32 @@ Please be sure the component that calls createSource or createToken is within an
 
       const elementOrDataResult = this.parseElementOrData(elementOrData);
 
-      // Second argument is Element; handleCardPayment with element
+      // Second argument is Element; handle with element
       if (elementOrDataResult.type === 'element') {
         const {element} = elementOrDataResult;
         if (maybeData) {
-          return stripe.handleCardPayment(clientSecret, element, maybeData);
+          return stripe[method](clientSecret, element, maybeData);
```
Comment (cweiss-stripe) [inline src/components/inject.js]: Not the biggest fan of dynamically calling methods like this, but I guess duplication/more nested if-else blocks isn't great either. 🤷‍♂

**Comment by cweiss-stripe** [comment]
lgtm

File: src/components/inject.js (hunk 1 above)
Comment (dweedon-stripe) [reply src/components/inject.js]: == but the underlying logic is was so identical that it felt really silly to duplicate everything. It is typed at least.

**Comment by dweedon-stripe** [comment]
cc @michelle-stripe

**Comment by paulhenrich-squarespace** [comment]
Just a heads-up, I noticed that the STRIPE_PUBLISHABLE_KEY doesn't actually get set right now, which breaks the demo if you set a STRIPE_SECRET_KEY. I was able to fix it by adding an additional entry to `webpack.config.js`:
```js
    new webpack.DefinePlugin({
      'process.env.STRIPE_PUBLISHABLE_KEY': JSON.stringify(
        process.env.STRIPE_PUBLISHABLE_KEY
      ),
    }),
```

**Comment by dweedon-stripe** [comment]
Ah thanks for catching that @paulhenrich-squarespace will fix.

**Pull request merged**